app/flatpak-builtins-run.c

```diff
@@ -168,7 +171,7 @@ flatpak_builtin_run (int argc, char **argv, GCancellable *cancellable, GError **
       if (runtime_ref == NULL)
         return FALSE;
 
-      runtime_deploy = flatpak_find_deploy_for_ref (runtime_ref, opt_commit ? opt_commit : opt_runtime_commit, cancellable, &local_error2);
+      runtime_deploy = flatpak_find_deploy_for_ref_in (dirs, runtime_ref, opt_commit ? opt_commit : opt_runtime_commit, cancellable, &local_error2);
```

Comment:
I'm not sure this is the right behaviour. This means that dependencies will *only* be taken from the specified repo. I guess it depends on what the usecase its.

Generally, --user always wins, so this would be used either to run the system version of an app, or to run an app installed from a different system installation that already is in the default system installation. In the first case I assume it seems likely that there would be a matching system runtime. However, in the second case it seems not as clear cut.

However, without massively complicating the CLI for this I don't really see a different way this could work...